csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -758,31 +764,42 @@ public sealed class Settings
             /// <summary>
             /// Default settings, as used by <see cref="JsonFormatter.Default"/>
             /// </summary>
-            public static Settings Default { get; }
+            public static Settings Default { get; } = new Settings();
 
-            // Workaround for the Mono compiler complaining about XML comments not being on
-            // valid language elements.
-            static Settings()
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object.
+            /// </summary>
+            public Settings() : this(false, TypeRegistry.Empty)
             {
-                Default = new Settings(false);
             }
 
             /// <summary>
             /// Whether fields whose values are the default for the field type (e.g. 0 for integers)
             /// should be formatted (true) or omitted (false).
             /// </summary>
-            public bool FormatDefaultValues { get; }
+            public bool FormatDefaultValues { get; set; }
```

Comment:
I really don't like making these settings mutable. This means that what one library does with the default settings will be visible to everything else in the application. Aside from anything else, changing something from being immutable to being mutable feels like a breaking change to me: users can no longer rely on something they used to be able to rely on.

If we think the settings are getting too complicated to have lots of constructors, we can introduce a builder, but let's keep the settings themselves immutable.

My preference would be to just add a third constructor, but make all the parameters optional:

```csharp
public Settings(bool formatDefaultValues = false, TypeRegistry typeRegistry = null, formatEnumAsInteger = false)
{
  // Normal code, using typeRegistry ?? TypeRegistry.Empty so that null means default
}
```

I realize that allowing a null value here but not elsewhere is inconsistent. Maybe we should change the other constructor to allow null as well. (That way feels like it's not a breaking change.)